cheatsheets/OS_Command_Injection_Defense_Cheat_Sheet.md

```diff
@@ -58,11 +84,12 @@ If calling a system command that incorporates user-supplied cannot be avoided, t
 - In regards to the **arguments** used for these commands, they should be validated using the following options:
     - **Positive or allow list input validation**: Where are the arguments allowed explicitly defined.
     - **Allow list Regular Expression**: Where a list of good, allowed characters and the maximum length of the string are defined. Ensure that metacharacters like ones specified in `Note A` and white-spaces are not part of the Regular Expression. For example, the following regular expression only allows lowercase letters and numbers and does not contain metacharacters. The length is also being limited to 3-10 characters: `^[a-z0-9]{3,10}$`
+- According to **Guidline 10** of this [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html), *The first -- argument that is not an option-argument should be accepted as a delimiter indicating the end of options. Any following arguments should be treated as operands, even if they begin with the '-' character.* For example, `curl -- $url` will prevent an argument injection even if the `$url` is malformed and contains an additional argument.
```

Comment:
Do keep in mind that it says _should_, **not** _must_. Also, even when intended, it often isn't implemented correctly.

Reply:
Agree on that. What do you think @h4r1337 should we change _should_ to _must_ here?

Reply:
@mackowski Actually that exact sentence was copied from the Guidline 10 of the POSIX page. And as @kwwall said it often isn't implemented correctly. And if it gives more clarity then I think it is better to change that to `must` instead of `should`.

Reply:
Should that be 'Guideline 10' instead of 'Guidline 10' or does POSIX just not know how to spell.

Reply:
No its my bad :sweat_smile: Fixing it...